### PR TITLE
Updated pom file to set Project description and URL

### PR DESCRIPTION
**Pull request by ocristian**
Setting the project description and URL to point to Github.

### Your checklist for this pull request

🚨 Please review the [guidelines for contributing](../blob/master/docs/CONTRIBUTING.md) to this repository.

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side) and not your master branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or in [Jenkins JIRA](https://issues.jenkins-ci.org)
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Did you provide a test-case? That demonstrates feature works or fixes the issue.

### CODE REVIEW

**Comment by casz** [comment]
this is for the parent pom, which will not be included in the plugin pom: https://github.com/jenkinsci/configuration-as-code-plugin/blob/7e04de3536cbf1970d9024d3f7662c061f422acf/plugin/pom.xml#L17-L18

File: pom.xml [hunk 1]
```diff
@@ -27,6 +27,8 @@
   </properties>
 
   <name>Configuration as Code Parent</name>
+  <description>The Configuration as Code plugin has been designed as an opinionated way to configure Jenkins based on human-readable declarative configuration files</description>
+  <url>https://github.com/jenkinsci/configuration-as-code-plugin</url>
```
Comment (casz) [inline pom.xml]: this is for the parent pom, which will not be included in the plugin pom: https://github.com/jenkinsci/configuration-as-code-plugin/blob/7e04de3536cbf1970d9024d3f7662c061f422acf/plugin/pom.xml#L17-L18

**Comment by oleg-nenashev** [comment]
+0 from me. It makes sense to have something in the root POM, but it may also cause confusion if somebody decides to improve the description later

**Comment by ocristian** [comment]
Hi @casz / @oleg-nenashev.
Thanks for your comments. 
Please let me know if there is anything else I can do here. 👍

**Comment by casz** [comment]
Perhaps remove the description on the plugin/pom so it will use the parent pom one?

**Comment by oleg-nenashev** [comment]
oops, I missed the last commit. It was a bad idea, metadata fields in pom.xml are not inherited from Parent POMs. It caused https://github.com/jenkinsci/configuration-as-code-plugin/issues/1206

**Comment by casz** [comment]
We could force it in the update center? :think

**Comment by oleg-nenashev** [comment]
You mean defaulting to README if nothing is specified?

**Comment by casz** [comment]
That is also a good option.
I was thinking more along the lines of https://github.com/jenkins-infra/update-center2/blob/master/src/main/resources/wiki-overrides.properties

**Pull request merged**